src/python/pants/backend/python/goals/setup_py.py

```diff
@@ -433,22 +429,40 @@ async def run_setup_py(req: RunSetupPyRequest, setuptools: Setuptools) -> RunSet
             interpreter_constraints=req.interpreter_constraints,
         ),
     )
+
+    # We prefix the entire chroot, and run with this prefix as the cwd, so that we can capture any
+    # changes setup made within it (e.g., when running 'develop') without also capturing other
+    # artifacts of the pex process invocation.
+    chroot_prefix = "chroot"
```

Comment:
It seems like we should be doing this more generally. Afaict no good ever comes of mixing binaries in with the sources they operate on at the top level in sandboxes.